packages/core/core/src/services/document-service/repository.ts

```diff
@@ -278,28 +287,35 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
       i18n.multiLocaleToLookup(contentType)
     )(params);
 
-    // Delete all drafts that match query
-    await deleteDocument({
-      ...queryParams,
-      documentId,
-      lookup: { ...queryParams?.lookup, publishedAt: null },
-    });
-
-    // Get deep populate of published versions
-    const entriesToDraft = await strapi.db?.query(uid).findMany({
-      where: {
-        ...queryParams?.lookup,
-        documentId,
-        publishedAt: { $ne: null },
-      },
-      populate: getDeepPopulate(uid, { relationalFields: ['documentId', 'locale'] }),
-    });
+    const [versionsToDraft, versionsToDelete] = await Promise.all([
```

Comment:
It's a bummer Promise.all doesn't seem to infer the types, I think you can pass a generic though if you think it's worth it.

Reply:
I thought Promise.all was smart enough 🤔 are you sure findMany returns the right types?

Reply:
strapi.db.query don't have return types not sure it's anything to do with Promise.all

Reply:
ah yeah you are right I think I misread something